### PR TITLE
17wf/tests news

### PR DESCRIPTION
**Pull request by Piwakk**

### CODE REVIEW

File: associations/models/news.py [hunk 1]
```diff
@@ -7,17 +7,18 @@
...
 
-    title = models.CharField(max_length=200, blank=True, null=True, default=None)
+    association = models.ForeignKey(Association, on_delete=models.CASCADE)
+    author = models.ForeignKey(User, on_delete=models.CASCADE)
 
     date = models.DateTimeField(auto_now=True)
-    author = models.ForeignKey(User, on_delete=models.CASCADE)
 
-    association = models.ForeignKey(Association, on_delete=models.CASCADE)
+    title = models.CharField(max_length=200, blank=True, null=True, default=None)
```
Comment (abocquet) [inline associations/models/news.py]: Maybe blank title shouldn't be authorized

File: associations/urls.py [hunk 2]
```diff
@@ -26,9 +25,10 @@
 
 associations_router = routers.NestedSimpleRouter(router, r'associations', lookup='association')
 associations_router.register(r'events', EventViewSet)
-#router.register(r'associations/(?P<association_id>[a-zA-Z0-9-_]+)/events', EventViewSet)
+associations_router.register(r'news', NewsAssociationViewSet)
 
 urlpatterns = [
+    path('news/', NewsSubscriptionsViewSet.as_view()),
```
Comment (abocquet) [inline associations/urls.py]: Are the news routes registered twice here ? (line 28 ans 31)

File: associations/views/news.py [hunk 3]
```diff
@@ -14,19 +15,32 @@ class TimelinePagination(PageNumberPagination):
...
-class NewsViewSet(viewsets.ModelViewSet):
+class NewsAssociationViewSet(AssociationNestedViewSet):
+    """This view deals with the news linked to a specific association."""
+
+    queryset = News.objects.all()
+    serializer_class = NewsSerializer
+    pagination_class = TimelinePagination
+    permission_classes = (NewsPermission,)
+
+    def get_queryset(self):
```
Comment (abocquet) [inline associations/views/news.py]: 👍

File: associations/views/news.py [hunk 4]
```diff
@@ -14,19 +15,32 @@ class TimelinePagination(PageNumberPagination):
...
+        return News.objects.filter(association=self.kwargs['association_pk'])
+
+    def perform_update(self, serializer):
```
Comment (abocquet) [inline associations/views/news.py]: 💯

File: associations/views/news.py [hunk 5]
```diff
@@ -14,19 +15,32 @@ class TimelinePagination(PageNumberPagination):
...
+        serializer.save(association=Association.objects.get(pk=self.kwargs['association_pk']))
+
+    def perform_create(self, serializer):
```
Comment (abocquet) [inline associations/views/news.py]: ❤️

File: associations/models/news.py (hunk 1 above)
Comment (Piwakk) [reply associations/models/news.py]: Indeed.

File: associations/urls.py (hunk 2 above)
Comment (Piwakk) [reply associations/urls.py]: No, one is for the `NewsAssociationViewSet`, the other for `NewsSubscriptionsViewSet` (which I renamed `NewsSubscriptionsView` by the way).

**Pull request merged**